### PR TITLE
tests: fix br_tikv_outage learner bug

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve? 

fix issue #1050 
In the case `outage-after-request`, there is a voter peer on tikv1 removed, and then a learner peer on tikv1 added.
But then quickly in the next case `outage-at-finegrained` (don't restart tikv while doing backup), tikv3 is killed.
So there is only 1 peer and 1 learner for region 257, lead to test got stuck.

### What is changed and how it works?

It seems the case `outage-after-request` have a bad influence on the next case, so this PR try to exchange the sequence of these cases.

### Check List 
Related changes

 - Need to cherry-pick to the release branch

### Release note

 -No release note

### CODE REVIEW

**Comment by Leavrth** [comment]
@YuJuncen PTAL

File: tests/br_tikv_outage/run.sh [hunk 1]
```diff
@@ -58,8 +58,9 @@ hint_finegrained=$TEST_DIR/hint_finegrained
 hint_backup_start=$TEST_DIR/hint_backup_start
 hint_get_backup_client=$TEST_DIR/hint_get_backup_client
 
-
-cases=${cases:-'outage outage-after-request outage-at-finegrained shutdown scale-out'}
+# NOTE : cases `outage-at-finegrained shutdown scale-out` should be former to avoid issue
```
Comment (kennytm) [inline tests/br_tikv_outage/run.sh:61]:
```suggestion
# NOTE : cases `outage-at-finegrained shutdown scale-out` should be first to avoid issue
```

**Review by kennytm — APPROVED** [review]

**Comment by glorv** [comment]
This sounds weird to me, I remember the scripts will recover the cluster to normal state before run each case, thus cases won't have side effect for each other. Are we sure this is the root cause? This case had already been fixed for several times 😭, is this the last time

**Comment by Leavrth** [comment]
> This sounds weird to me, I remember the scripts will recover the cluster to normal state before run each case, thus cases won't have side effect for each other. Are we sure this is the root cause? This case had already been fixed for several times 😭, is this the last time

This test has many bugs which all can make test failed, And I think this is the last bug. 
In fact, the recover in scripts only make sure that the cluster is already initialized before running each case. (This is fixed in other PR #1160 )
As shown above, even though this killed node has been restarted and initialized, there are still some peers in this node hasn't been synchronized with other peers in other nodes, which make these peers keep `learner state`.
At this point, if we kill another node, these `learner state` peers might be failed to synchronize itself, which makes the test timeout.

**Review by glorv — APPROVED** [review]

**Pull request merged**